### PR TITLE
organize golangci workflow

### PR DESCRIPTION
**Pull request by mmorel-35**
# Please add a summary of your change

* check out before go setup 
* use go version from go.mod
* define output format in golangci 

# Does your change fix a particular issue?

Fixes #(issue)

# Please indicate you've done the following:

- [x] [Accepted the DCO](https://velero.io/docs/v1.5/code-standards/#dco-sign-off). Commits without the DCO will delay acceptance.
- [x] [Created a changelog file](https://velero.io/docs/v1.5/code-standards/#adding-a-changelog) or added `/kind changelog-not-required` as a comment on this pull request.
- [ ] Updated the corresponding documentation in `site/content/docs/main`.

### CODE REVIEW

**Comment by kaovilai** [comment]
conflicts with https://github.com/vmware-tanzu/velero/pull/7194

**Review by mateusoliveira43 — APPROVED** [review]
I can rebase my PR afterwards

**Review by kaovilai — APPROVED** [review]

**Review by blackpiglet — APPROVED** [review]

**Review by blackpiglet — DISMISSED** [review]

**Review by kaovilai — APPROVED** [review]

**Comment by anshulahuja98** [comment]
CI failed due to throttling, rerunning it.

**Review by kaovilai — APPROVED** [review]

**Review by anshulahuja98 — DISMISSED** [review]

**Review by blackpiglet — CHANGES_REQUESTED** [review]
I suggest not to change the configuration file name.
I'm trying to use `golangci/golangci-lint-action@v4` to replace the `make lint` in the lint action.
And the GitHub action cannot find files those names start with `.`.

**Comment by blackpiglet** [comment]
> .golangci.yaml is one of the default file handled by golangci-lint so you won't need to customize the golangci config file, it will work out of the box, I can complete this PR to package it if you want ?

Please take a view of this commit. 
https://github.com/vmware-tanzu/velero/pull/7697/commits/abb1140c201a03124399f5f232da11919a242138

The linter's output is not formatted if it doesn't pass the configuration file parameter.
If the configuration file parameter is specified, the action always complains ".golangci.yaml" cannot be found.

**Review by blackpiglet — DISMISSED** [review]

File: .github/workflows/pr-linter-check.yml
Comment (blackpiglet) [reply .github/workflows/pr-linter-check.yml:16]: Great. That would be very helpful.

**Review by blackpiglet — APPROVED** [review]

File: .github/workflows/pr-linter-check.yml
Comment (blackpiglet) [reply .github/workflows/pr-linter-check.yml:16]: Sure.

**Review by kaovilai — APPROVED** [review]

**Review by kaovilai — APPROVED** [review]

**Review by blackpiglet — DISMISSED** [review]

**Comment by blackpiglet** [comment]
We need the maintainers' approval to merge the PR.
Involve more people to accelerate the process.

**Review by blackpiglet — APPROVED** [review]

**Review by sseago — APPROVED** [review]

**Pull request merged**
